Documentation/ceph-dashboard-monitoring.md

```diff
@@ -0,0 +1,291 @@
+
+In order to deploy Grafana properly configured for Ceph, we will use the grafana.yaml file stored in the monitoring folder. Some modifications in this file could be needed: 
+
+**1. Change of SSL certificates (optional)**
+
+The grafana.yaml file includes a configmap for the SSL certificates used to enable secure access to the UI. These certificates are self-signed and if it is needed or preferred can be replaced by others.
+
+**2. Grafana storage (mandatory)**
+
+Grafana needs storage to save database and other stuff. The "grafana.yaml" file provided uses a PVC that get the storage from a PV based in a Storage Class with name "ceph-storage". The user must be sure that a PV using this storage class is available for the grafana PVC. Or change the storage class used in the grafana PVC to point the right one for your cluster. 
+
+Verify you have a PV with enough space(at least 500Mb) available for the Grafana PVC:
+```console
+$ kubectl get pv
+NAME              CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM   STORAGECLASS   REASON   AGE
+ceph-pv-grafana   9Gi        RWO            Retain           Available           ceph-storage            5s
+```
+
+**3. Modify the grafana.ini file to point the right Prometheus service URL (mandatory)**
+
+Go to the monitoring folder (where the grafana.yaml file is located) and execute:
+```console
+$ sed -i "s/PROMETHEUS-SVC/http:\/\/$(kubectl -n rook-ceph -o jsonpath={.status.hostIP} get pod prometheus-rook-prometheus-0):30900/g" grafana.yaml
+```
+
+Once we have updated the grafana ini file with the Prometheus URL, and we know that there is A PV with enough space available for Grafana, we can proceed to create the Grafana deployment and service:
+```console
+$ kubectl create -f grafana.yaml
+```
+To verify Grafana is up and running, just check deployments and services in the rook-ceph workspace. To access the web interface, execute:
+
+```console
+$ echo "https://$(kubectl -n rook-ceph -o jsonpath='{.items[0].status.hostIP}' get pod --selector=app=grafana):32000"
+https://192.168.122.157:32000
+```
+
+Which will give you the url to use in your browser to see the Grafana UI.
+
+![Grafana home page](media/grafana-home.png)
+
+Once you have an instance of Grafana connected with Prometheus, you will need to connect Grafana with the Ceph Dashboard following the next steps: 
+
+   1. In one of your Kubernetes nodes, obtain the grafana URL:
+
+      ```console
+      $ echo "https://$(kubectl -n rook-ceph -o jsonpath='{.items[0].status.hostIP}' get pod --selector=app=grafana):32000"
+      https://192.168.122.157:32000
+      ```
+   2. Open a Ceph toolbox and provide the Grafana URL to the Ceph Dashboard
+
+      ```console
+      [root@rook-ceph-tools-78cdfd976c-8xvgn /]# ceph dashboard set-grafana-api-url https://192.168.122.157:32000
```

Comment:
```suggestion
      # ceph dashboard set-grafana-api-url https://192.168.122.157:32000
```
In general, we keep  `$` or `#` in front of command where we can't directly copy command and run. 
example:
In above command URL may be different sometime so we keep `$` or`# ` in front but for `kubectl get pv` we don't need to have `$` or `#` in front.